homeassistant/components/snmp/__init__.py

```diff
@@ -1 +1,44 @@
 """The snmp component."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant import config as conf_util
+from homeassistant.core import Event
+from homeassistant.const import SERVICE_RELOAD
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.reload import async_reload_integration_platforms
+from homeassistant.loader import async_get_integration
+
+from .const import DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up the snmp integration."""
+
+    async def _reload_config(call: Event) -> None:
+        """Reload top-level + platforms."""
+        try:
```

Comment:
We should let the exception propagate so feedback is given to the frontend what the error is